src/image.py

```diff
@@ -69,34 +72,67 @@ def collect_installed_packages_information(self):
         docker_client.containers.prune()
         return command_responses
 
+    def pre_build_configuration(self):
+
+        if self.info.get("base_image_uri"):
+            self.build_args["BASE_IMAGE"] = self.info["base_image_uri"]
+
+        if self.ecr_url:
+            self.build_args["INITIAL_STAGE_IMAGE"] = self.ecr_url
+
+        if self.info.get("extra_build_args"):
+            self.build_args.update(self.info.get("extra_build_args"))
+        
+        if self.info.get("labels"):
+            self.labels.update(self.info.get("labels"))
+        
+        print(f"self.build_args {self.build_args}")
+        print(f"self.labels {self.labels}")
+
     def build(self):
         """
         The build function builds the specified docker image
         """
         self.summary["start_time"] = datetime.now()
 
+        ## Confirm if building the image is required or not
         if not self.to_build:
             self.log = ["Not built"]
             self.build_status = constants.NOT_BUILT
             self.summary["status"] = constants.STATUS_MESSAGE[self.build_status]
             return self.build_status
+        
+        ## Conduct some preprocessing before building the image
+        self.pre_build_configuration()
+        print(f"self.context {self.context}")
+
+        ## Start building the image
+        if self.context:
+            with open(self.context.context_path, "rb") as context_file:
+                print("within context")
+                self.docker_build(fileobj=context_file, custom_context=True)
+                self.context.remove()  
+        else:
```

Comment:
Is there a case where the `self.docker_build()` is invoked?

Reply:
I haven't seen any yet. It was already existing in the code, so I thought of not playing with it.

Reply:
Existing code as in on the PR branch or the master, if PR branch then lets just remove it, probably it might have been added for experimental purposes.

Reply:
Sure, I will remove it.